src/v2/System/Relay/SystemQueryRenderer.tsx

```diff
@@ -1,34 +1,52 @@
 import React from "react"
-// FIXME: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/37950
-// @ts-ignore
-import { QueryRenderer, QueryRendererProps } from "react-relay"
+import { useEffect } from "react"
+import { useState } from "react"
+import { QueryRenderer } from "react-relay"
 import { OperationType } from "relay-runtime"
 
-interface SystemQueryRendererState {
-  isMounted: boolean
+type QueryRendererProps = React.ComponentProps<typeof QueryRenderer>
+
+export type SystemQueryRendererProps<T extends OperationType> = Omit<
+  QueryRendererProps,
+  "render" | "environment" | "variables"
+> & {
+  placeholder?: React.ReactNode
+  environment?: QueryRendererProps["environment"]
+  variables?: QueryRendererProps["variables"]
+  render(renderProps: {
+    error: Error | null
+    props: T["response"] | null
+    retry: (() => void) | null
+  }): React.ReactNode
```

Comment:
This is a bit annoying — the only way I was able to figure out how to thread the response props from generated types through was by manually re-typing `render` — if anyone has suggestions on how to pass this into `QueryRendererProps` above, I'm all ears. I'm probably missing something obvious.